### ISSUE TITLE
Make the Entanglement component API sugar

### ISSUE THREAD

**Issue by pirelenito** [react-entanglement/react-entanglement#6]
We can prioritize having the `communicationAdapter` passed as arguments to the `materialize` and `scatter` HOCs, and if the user wants... they can use the Entanglement component to set it in the context.

**Comment by pirelenito** [comment]
So the idea is that we are ditching the HOCs, but we will support both having the `adapter` available in the `context` or defined locally as a prop of either the `Materialize` or `Scatter` components:

So you could have the same result using either the `context`:

``` js
// Main App
<Entanglement adapter={adapter}>
  <div>
    <Scatter name='Dialog' props={props} />
  </div>
</Entanglement>

// Remote App
<Entanglement adapter={adapter}>
  <div>
    <Materialize name='Dialog' component={Dialog} />
  </div>
</Entanglement>
```

Or directly in the components:

``` js

// Main App
<div>
  <Scatter name='Dialog' props={props} adapter={adapter} />
</div>

// Remote App
<div>
  <Materialize name='Dialog' component={Dialog} adapter={adapter} />
</div>
```

We've also decided to not have the `Scatter` component fallback to the default implementation if the `adapter` is not available. This should be handled by the user! If the adapter is not available we shall give a warning!

If there is conflicting adapters, either component/wrapper or wrapper/wrapper we should warn as well.

This was elaborated in conjunction with @xaviervia in a real room! Not in the interwebs... 😳

**Comment by pirelenito** [comment]
Was fixed in #9
